www/src/components/Menu.tsx

```diff
@@ -49,9 +52,15 @@ const menuQuery = graphql`
   }
 `;
 
+export interface ComponentNavItemPropsTypes {
+  id: string,
+  fields: PropTypes.InferProps<{ slug: string }>,
```

Comment:
[curious] What is `PropTypes.InferProps` here?

Reply:
With `PropTypes.InferProps` we can extract the type from the PropTypes object